torch/_ops.py

```diff
@@ -409,6 +430,93 @@ def key_extractor(tensors, key_mask):
 _mode_stack_per_key: Dict[torch._C.DispatchKey, List] = {}
 
 
+class ModeSlot(enum.IntEnum):
```

Comment:
nit: I'm not sure that this new enum buys us much over just using the existing `TorchDispatchModeKey` enum, and asserting that fake is never in it (since this introduces an extra layer of indirection)

Reply:
deleted